Core/KSPManager.cs

```diff
@@ -151,6 +152,76 @@ public KSP AddInstance(KSP ksp_instance)
             return ksp_instance;
         }
 
+        /// <summary>
+        /// Clones an existing KSP installation.
+        /// </summary>
+        /// <param name="existing_instance">The KSP instance to clone.</param>
+        /// <param name="new_name">The name for the new instance.</param>
+        /// <param name="new_path">The path where the new instance should be located.</param>
+        public void CloneInstance (KSP existing_instance, string new_name, string new_path)
+        {
+            if (existing_instance.Valid)
+            {
+                KspVersion version = existing_instance.Version();
+
+                CKAN.DLC.MakingHistoryDlcDetector dlcDetector = new DLC.MakingHistoryDlcDetector();
+                bool DLC = dlcDetector.IsInstalled(existing_instance, out string identifier, out UnmanagedModuleVersion moduleVersion);
+                string dlcVersion = moduleVersion.ToString();
+
+                FakeInstance(new_name, new_path, version, DLC, dlcVersion);
+            }
+            else
+            {
+                NotKSPDirKraken kraken = new NotKSPDirKraken(existing_instance.GameDir());
+                log.Error(kraken);
+                throw kraken;
+            }
+        }
+
+        /// <summary>
+        /// Create a new fake KSP instance
+        /// </summary>
+        /// <param name="new_name">The name for the new instance.</param>
+        /// <param name="new_path">The loaction of the new instance.</param>
+        /// <param name="version">The version of the new instance.</param>
+        /// <param name="DLC">Whether to fake the DLC too.</param>
+        /// <param name="dlcVersion">The version of the DLC. Can be null if DLC == false.</param>
+        public void FakeInstance(string new_name, string new_path, KspVersion version, bool DLC, string dlcVersion)
+        {
+            try
+            {
+                if (KSP.IsKspDir(new_path)) {
+                    throw new BadInstallLocationKraken("There is already a KSP instance at this path. Delete the old one first.");
+                }
+
+                log.DebugFormat("Creating folder structure and text files at {0} for KSP version {1}", Path.GetFullPath(new_path), version.ToString());
+
+                // Create a KSP root directory, containing a GameData folder, a buildID.txt and a readme.txt
+                Directory.CreateDirectory(new_path);
+                Directory.CreateDirectory(Path.Combine(new_path, "GameData"));
+                File.WriteAllText(Path.Combine(new_path, "buildID.txt"), String.Format("build id = {0}", version.Build));
```

Comment:
`version.Build` could still be null here, e.g. if I choose a fake version like "1.2.4". You'll probably have to do an explicit `IsBuildDefined` check and skip this step if false.

Reply:
`AddBuildToVersion()` returns a  null object if it can't find the version in the build map, and therefore `FakeNewKSPInstall` throws an exception before `FakeInstace()` is called.
But if we call it from anywhere else in the future, it might make sense to to a check inside this function.

Reply:
Thinking more about this, it would be useful to have a new `public bool IsValid()` in `KspVersion`.
That could be checked in `FakeInstance()` and if `false`, throw an exception.
I don't see a sense in allowing unreal KSP versions in a  dummy instance.